src/warp.jl

```diff
@@ -124,8 +124,28 @@ julia> imrotate(img, π/4, Constant())
 
 See also [`warp`](@ref).
 """
-function imrotate(img::AbstractArray{T}, θ::Real, args...) where T
-    θ = floor(mod(θ,2pi)*typemax(Int16))/typemax(Int16) # periodic discretezation
-    tform = recenter(RotMatrix{2}(θ), center(img))
-    warp(img, tform, args...)
+function imrotate(img::AbstractMatrix{T}, θ::Real, args...) where T
+    # 1. discretize periodic for numerical stability to make sure
+    # imrotate(img, θ+2pi) == imrotate(img, θ)
+    # 2. typemax(Int16) is 32767, we choose 32760 to make sure the
+    # discretization result of pi/2 is exactly pi/2 (or 90°)
+    max_num_angles = 32760
```

Comment:
Although the same goes for the current code, why don't you use 32768?

Reply:
Well... It's like I randomly choose the first number that meets the requirement, and I feel this choice doesn't matter for image processing tasks.